dallinger/frontend/static/scripts/tracking/scribe-dallinger.js

```diff
@@ -0,0 +1,42 @@
+var ScribeDallingerTracker = function(config) {
+  if (!(this instanceof ScribeDallingerTracker)) return new ScribeDallingerTracker(config);
+
+  this.config = config;
+};
+
+ScribeDallingerTracker.prototype.tracker = function(info) {
+  var config = this.config;
+  var path = info.path;
+  var value = info.value || {};
+  var data = new FormData(); 
+  if (config.base_url) {
+    // Only track events
+    if (path.indexOf('/events/') < 0) {
+      return;
+    }
+
+    // Remove possible PII
+    if (value.fingerprint) {
+      delete value.fingerprint;
+    }
+    if (value.visitorId) {
+      delete value.visitorId;
+    }
+
+    data.append('Event.1.EventType', 'TrackingEvent');
+    if (config.participant_id) data.append('participant_id', config.participant_id);
+    if (config.assignment_id) data.append('Event.1.AssignmentId', config.assignment_id);
+    data.append('details', JSON.stringify(value));
+
+    var xhr = new XMLHttpRequest();
+    if (info.success) xhr.addEventListener("load", info.success);
+    if (info.failure) {
+      xhr.addEventListener("error", info.failure);
+      xhr.addEventListener("abort", info.failure);
+    }
+    xhr.open('POST', config.base_url.replace(/\/$/, "") + '/notifications', true);
+    xhr.send(data);
+  } else {
+    if(info.failure) setTimeout(info.failure, 0);
```

Comment:
Why the setTimeout with a delay of 0?

Reply:
I have no idea, that's what the scribe code did in its console tracker sample integration and that bit was a copy paste. Seems strange, but I just left it as it was. Frankly, I don't think there's any case in our code where `info.failure` is set, so this code path is probably entirely vestigial.